src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -625,7 +625,9 @@ JvmtiEnvBase::get_field_descriptor(Klass* k, jfieldID field, fieldDescriptor* fd
 
 bool
 JvmtiEnvBase::is_vthread_alive(oop vt) {
-  return java_lang_VirtualThread::state(vt) != java_lang_VirtualThread::NEW &&
+  oop cont = java_lang_VirtualThread::continuation(vt);
+  return !jdk_internal_vm_Continuation::done(cont) &&
+         java_lang_VirtualThread::state(vt) != java_lang_VirtualThread::NEW &&
          java_lang_VirtualThread::state(vt) != java_lang_VirtualThread::TERMINATED;
```

Comment:
AFAIU `jdk_internal_vm_Continuation::done(cont)` is correct check that vthread is terminated and works for both mounted and unmounted vthreads.
Then `java_lang_VirtualThread::state(vt) != java_lang_VirtualThread::TERMINATED` check is not needed

Reply:
Good suggestion, thanks!

Reply:
Added the fix suggested by Alex.